packages/ember-runtime/lib/copy.js

```diff
@@ -19,12 +21,12 @@ function _copy(obj, deep, seen, copies) {
 
   assert(
     'Cannot clone an EmberObject that does not implement Copyable',
-    !(obj instanceof EmberObject) || Copyable.detect(obj)
+    !(obj instanceof EmberObject) || Copyable.detect(obj) || NativeArray.detect(obj)
   );
 
   // IMPORTANT: this specific test will detect a native array only. Any other
   // object will need to implement Copyable.
-  if (Array.isArray(obj)) {
+  if (Array.isArray(obj) || NativeArray.detect(obj)) {
```

Comment:
we only apply the NativeArray mixin to arrays, so Array.isArray should be enough.